src/client.ts

```diff
@@ -94,6 +85,27 @@ export class Client {
     this._assignedRoute = null;
   }
 
+  /**
+   * Registers custom elements used by the client application
+   */
+  public registerCustomElements(): void {
+    const clientInstance = this;
+    class IfcClientLinkElementComplete extends IfcClientLinkElement {
```

Comment:
@MattCheely Changing the `registerCustomElements` standalone function to a method on the `Client` class solved the issues I was facing with being unable to access the environmentData. But when I tried to commit my changes, I got a linting error `A maximum of 1 class per file is allowed.`. Did I implement this incorrectly?

Reply:
Not at all. Normal best practice is one class per file, but that's working under the assumption that you're making classes for external consumption, or that the classes are complex. This is a special case in that we need to provide a class that can close over a local value. 

I'd just add a [tslint disable comment](https://palantir.github.io/tslint/usage/rule-flags/) for that rule on the line it's erroring on, presumably this one.